wire.go

```diff
@@ -47,6 +47,7 @@ type ProviderSet struct{}
 // Passing a struct value of type S to NewSet declares that both S and *S will
 // be provided by creating a new value of the appropriate type by filling in
 // each field of S using the provider of the field's type.
+// Deprecated: use wire.Struct.
```

Comment:
Let's deemphasize this paragraph by moving it to the end of the doc comment.

Also, I think we could frame the deprecation to give the reader more information with slightly different wording. I propose:

**For compatibility with older versions of Wire,** passing a struct value of type S to NewSet declares that both S and *S will be provided by creating a new value of the appropriate type by filling in each field of S using the provider of the field's type. **This form is deprecated and will be removed in a future version of Wire: new providers sets should use wire.Struct.**

Reply:
Done.